include/javabind/enum.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * javabind: effective C++ and Java interoperability
+ * @see https://github.com/hunyadi/javabind
+ *
+ * Copyright (c) 2024 Levente Hunyadi
+ *
+ * This work is licensed under the terms of the MIT license.
+ * For a copy, see <https://opensource.org/licenses/MIT>.
+ */
+
+#pragma once
+#include "exception.hpp"
+#include "message.hpp"
+#include "object.hpp"
+#include "signature.hpp"
+#include <algorithm>
+#include <optional>
+#include <type_traits>
+
+namespace javabind
+{
+    template <typename native_type>
+    struct EnumValues
+    {
+        inline static std::map<native_type, std::string_view> values_to_names;
+        inline static std::map<std::string_view, native_type> names_to_values;
```

Comment:
I assume ordinals assigned to enumeration values in Java are contiguous. If so, we could employ an `std::vector<native_type>` to map Java enumeration ordinals to C++ enumeration values.

Mapping C++ enumeration values to Java enumeration objects could involve a hash map for *O(1)* retrieval.

Reply:
They are contiguous but not every field of the Java enum may be bound. I changed the map to an unordered_map to have O(1) time complexity for retrieval.